src/com/esotericsoftware/kryonet/TcpConnection.java

```diff
@@ -200,9 +204,21 @@ public int send (Connection connection, Object object) throws IOException {
 		SocketChannel socketChannel = this.socketChannel;
 		if (socketChannel == null) throw new SocketException("Connection is closed.");
 		synchronized (writeLock) {
+			int lengthLength = serialization.getLengthLength();
+
+			//wait while buffer has enough free space.
+			//thats because message length will be put only after the whole message will be serialized
+            while (writeBuffer.capacity() < writeBuffer.position() + lengthLength + objectBufferSize) {
+                try {
+                    Thread.sleep(10);
+                } catch (InterruptedException e) {
+                    break;
+                }
+                writeToSocket();
+	        }
```

Comment:
If you pass 0 as value of objectSize, feature will be disabled. We can set 0 as default value.

Concerning "The right way", it's the matter of dispute. If so user should select unique buffer size for each system installation on different computers, in different configurations and so on. Or make buffer too large to avoid problems that are very rare.